process/transaction/shardProcess.go

```diff
@@ -789,6 +783,26 @@ func (txProc *txProcessor) processInnerTx(
 
 	txFee := txProc.computeInnerTxFee(innerTx)
 
+	innerTxHash := originalTxHash
+
+	var err error
+	if txProc.enableEpochsHandler.IsFlagEnabled(common.HashForInnerTransactionFlag) {
+		innerTxHash, err = core.CalculateHash(txProc.marshalizer, txProc.hasher, innerTx)
+		if err != nil {
+			return txFee, vmcommon.UserError, txProc.executeFailedRelayedUserTx(
+				innerTx,
+				innerTx.RelayerAddr,
+				big.NewInt(0),
+				tx.Nonce,
+				tx,
+				originalTxHash,
+				err.Error(),
+				innerTxIdx)
+		}
+
+		txProc.relayedTxV3InnerHashes = append(txProc.relayedTxV3InnerHashes, innerTxHash)
```

Comment:
in case of a v1 or v2 tx, those this is not a relayedV3 inner hash

Reply:
this method should be called for v3 only